### PR TITLE
feat: enable auto-refresh for kill-process

### LINKED ISSUE #6941: [Kill Process] Auto-Refresh like System Monitor
### Extension

https://www.raycast.com/rolandleth/kill-process

### Description

for now, Kill Process is only fetching the processes once while launch the command, and it is only able to refresh the list manually. I hope it could support auto-fresh like system monitor.

Thanks.

### Who will benefit from this feature?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by zhenpewu**
## Description

Closes #6941

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by zhenpewu** [comment]
#6941

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Thanks for your contribution @zhenpewu 🔥

I gave it a quick spin and it works as expected

Two things:

- Maybe make this optional with a checkbox in preferences?
- Could you add a changelog entry

@rolandleth do you mind checking this too?

File: extensions/kill-process/src/index.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-  const shouldShowPID = (preferences.shouldShowPID?.value as boolean) ?? false;
-  const shouldShowPath = (preferences.shouldShowPath?.value as boolean) ?? false;
+
+  const preferences = getPreferenceValues<Preferences>();
+  const shouldIncludePaths = preferences.shouldSearchInPaths ?? false;
+  const shouldIncludePid = preferences.shouldSearchInPid ?? false;
+  const shouldPrioritizeAppsWhenFiltering = preferences.shouldPrioritizeAppsWhenFiltering ?? false;
+  const shouldShowPID = preferences.shouldShowPID ?? false;
+  const shouldShowPath = preferences.shouldShowPath ?? false;
+  const refreshDuration = +preferences.refreshDuration;
```
[the remark this answers is not in the document]
Comment (zhenpewu) [reply extensions/kill-process/src/index.tsx:16]: sure, make sense to me.

**Comment by zhenpewu** [comment]
> Maybe make this optional with a checkbox in preferences?

Can I use -1 as disable ?

File: extensions/kill-process/package.json [hunk 2]
```diff
@@ -57,25 +58,35 @@
       "required": false,
       "title": "",
       "type": "checkbox"
+    },
+    {
+      "description": "Refresh duration in milliseconds. The minimum value is 1000ms and to disable auto-refresh, set it to -1.",
```
[the remark this answers is not in the document]
Comment (zhenpewu) [reply extensions/kill-process/package.json:63]: okay.

File: extensions/kill-process/src/index.tsx [hunk 3]
```diff
@@ -39,9 +44,7 @@ export default function ProcessList() {
     });
   };
 
-  useEffect(() => {
-    fetchProcesses();
-  }, []);
+  refreshEnabled ? useInterval(fetchProcesses, refreshDuration) : useEffect(fetchProcesses, []);
```
[the remark this answers is not in the document]
Comment (zhenpewu) [reply extensions/kill-process/src/index.tsx]: agree, updated useInterval.

File: extensions/kill-process/src/index.tsx [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
-  const shouldShowPID = (preferences.shouldShowPID?.value as boolean) ?? false;
-  const shouldShowPath = (preferences.shouldShowPath?.value as boolean) ?? false;
+
+  const preferences = getPreferenceValues<Preferences>();
+  const shouldIncludePaths = preferences.shouldSearchInPaths ?? false;
+  const shouldIncludePid = preferences.shouldSearchInPid ?? false;
+  const shouldPrioritizeAppsWhenFiltering = preferences.shouldPrioritizeAppsWhenFiltering ?? false;
+  const shouldShowPID = preferences.shouldShowPID ?? false;
+  const shouldShowPath = preferences.shouldShowPath ?? false;
+  const refreshEnabled = +preferences.refreshDuration >= 0;
```
[the remark this answers is not in the document]
Comment (pernielsentikaer) [reply extensions/kill-process/src/index.tsx]: The ﻿+ in front of a string is a unary operator that can be used to convert a string to a number. For example, ﻿+'42' would evaluate to the number ﻿42. If the string cannot be converted to a number, the result will be ﻿NaN (Not a Number).

**Comment by pernielsentikaer** [comment]
Still works as expected for me , could you verify @rolandleth

**Comment by rolandleth** [comment]
Code-wise looks good, so if it also works good, we're good to go. Good, right? 😅

**Pull request merged**